source/transactions/tests/run-command.yml

```diff
@@ -0,0 +1,167 @@
+              $numberLong: "1"
+            startTransaction:
+            autocommit: false
+            writeConcern:
+          command_name: commitTransaction
+          database_name: admin
+
+  - description: run command with secondary read preference in client option and primary read preference in transaction options
+
+    clientOptions:
+      readPreference: secondary
+
+    operations:
+      - name: startTransaction
+        arguments:
+          session: session0
+          options:
+            readPreference:
+              mode: Primary
+      - name: runCommand
+        arguments:
+          command: 
+            insert: *collection_name
+            documents:
+              - _id : 1
+          session: session0
+        result:
+          n: 1
+      - name: commitTransaction
+        arguments:
+          session: session0
+
+    expectations:
+      - command_started_event:
+          command:
+            insert: *collection_name
+            documents:
+              - _id : 1
+            readConcern:
+            lsid: session0
+            txnNumber:
+              $numberLong: "1"
+            startTransaction: true
+            autocommit: false
+            writeConcern:
+          command_name: insert
+          database_name: *database_name
+      - command_started_event:
+          command:
+            commitTransaction: 1
+            lsid: session0
+            txnNumber:
+              $numberLong: "1"
+            startTransaction:
+            autocommit: false
+            writeConcern:
+          command_name: commitTransaction
+          database_name: admin 
+
+  - description: run command with explicit primary read preference
+
+    operations:
+      - name: startTransaction
+        arguments:
+          session: session0
+      - name: runCommand
+        arguments:
+          command: 
+            insert: *collection_name
+            documents:
+              - _id : 1
+          session: session0
+          readPreference:
+            mode: Primary
+        result:
+          n: 1
+      - name: commitTransaction
+        arguments:
+          session: session0
+
+    expectations:
+      - command_started_event:
+          command:
+            insert: *collection_name
+            documents:
+              - _id : 1
+            readConcern:
+            lsid: session0
+            txnNumber:
+              $numberLong: "1"
+            startTransaction: true
+            autocommit: false
+            writeConcern:
+          command_name: insert
+          database_name: *database_name
+      - command_started_event:
+          command:
+            commitTransaction: 1
+            lsid: session0
+            txnNumber:
+              $numberLong: "1"
+            startTransaction:
+            autocommit: false
+            writeConcern:
+          command_name: commitTransaction
+          database_name: admin
+
+  - description: run command fails with explicit secondary read preference
```

Comment:
Can you add another test like this but put the secondary read preference on startTransaction?

Reply:
Done